### DISCUSSION TITLE
Copilot with Live Share for Remote Pair Programming Supporting Disabled Developers

### DISCUSSION THREAD

**Discussion by Jim-Salmons** [community/community#13737] [Copilot Conversations]
I have been quiet but very active evolving the research design, data model, PI/funder network, and—as discussed briefly here—exploring the Copilot-enhanced Remote Pair Programming platform for the proposed “GitHub Copilot for Disabled Develops” disability research study and support program. I will provide a brief update of my other activities in a separate discussion. ITMT, here is some brief feedback on the good-but-could-be-better exploration of using Copilot with the Live Share extension to support Remote Pair Programming.

My brief initial experiment was limited to doing a VS Code Live Share session between native VSC on MacOS and VSC under Windows 11 on a Windows 11 VM via Parallels Desktop on an M1 iMac. As the figure below shows, this was easily set up and worked in an as-installed extension configuration. Copilot suggestions are shown to the Pair’s “driver” and become visible to the “navigator” once accepted by the driver. It would be great if the suggestions could be shown interactively to both Pair members, but this limitation can be overcome by the team’s use of Live Share Audio to discuss the driver-seen suggestion before acceptance. Once accepted and seen by both Pair members, either developer can quickly undo or edit the inserted Copilot suggested code.

I would be happy to hear from other Copilot users who have Remote Pair Programming experience. If you would like more information on the “GitHub Copilot for Disabled Developers” study/support project, please see this publication: https://medium.com/github-copilot-for-disabled-developers. I especially want to bring this discussion to the attention of @ryanjsalva & @irealva and members of the Copilot development team.

<img width="1120" alt="LiveShare_Copilot_suggestions" src="https://user-images.githubusercontent.com/546022/160520639-33352590-00db-4fb7-a2cf-201335d72715.png">

**Comment by vicdotdevelop** [comment]
+1

**Comment by coderforlife** [comment]
In many uses of Paired Programming, the individuals swap roles periodically. With the current setup of Copilot only being visible to the host, swapping requires stopping the liveshare and restarting it, an unnecessary distraction. Having them visible on both ends would allow swapping roles MUCH easier.
